### ISSUE TITLE
'OrderPayment.order' has not been mapped

### ISSUE THREAD

**Issue by gtolivier** [awesto/django-shop#477]
Hi,

I am trying to install Django Shop by hand.

I first installed and configured Django CMS which is working fine. Then I installed Django Shop with pip along with its dependencies and the required apps and some of the recommended ones. I configured it following the steps on that page: http://django-shop.readthedocs.io/en/latest/reference/configuration.html

But I am getting the following error when I try to migrate:

> django.core.exceptions.ImproperlyConfigured: Deferred foreign key 'OrderPayment.order' has not been mapped

Here is the content of my settings.py file:
[settings.py.txt](https://github.com/awesto/django-shop/files/625497/settings.py.txt)

Here are the apps I installed:

> beautifulsoup4==4.5.1
> Django==1.9.10
> django-admin-sortable2==0.6.6
> django-angular==0.8.4
> django-appconf==1.0.2
> django-classy-tags==0.8.0
> django-cms==3.3.3
> django-filer==1.2.5
> django-formtools==1.0
> django-fsm==2.4.0
> django-fsm-admin==1.2.4
> django-ipware==1.1.6
> django-mptt==0.8.6
> django-polymorphic==0.8.1
> django-post-office==2.0.8
> django-rest-auth==0.8.2
> django-sekizai==0.10.0
> Django-Select2==5.8.9
> django-shop==0.9.3
> django-treebeard==4.1.0
> djangocms-admin-style==1.2.6.2
> djangocms-cascade==0.11.0
> djangocms-text-ckeditor==3.3.1
> djangorestframework==3.5.3
> easy-thumbnails==2.3
> enum34==1.1.6
> html5lib==0.9999999
> jsonfield==1.0.3
> Pillow==3.4.2
> psycopg2==2.6.2
> six==1.10.0
> Unidecode==0.4.19

Do you have any idea how to solve this issue?

Do I have to code the models of my shop implementation first?
Do I have to copy the migration file of one of your examples to my shop implementation and customize it?

Thanks,

Olivier

**Comment by jrief** [comment]
For deferred foreign keys to work, Django must "see" the materialized model at least once.
You presumably never imported ``OrderPayment`` into your project. A good place to include this is via registering ``OrderAdmin`` into your admin interface.

If that helps, please let me know and so that I can close this issue.

**Comment by gtolivier** [comment]
I managed to make it work.
I thought I could customize my implementation shop later but as you said I needed to implement the materialized models first.

Thanks a lot!

**Comment by jrief** [comment]
Thank for the feedback. I'll close it.

**Comment by fabiancastano** [comment]
I'm new to Django, I'm a beginner.

But for personal purposes we want to make a web application with cms-django and django-shop integrate, for after this integration: the development of end user interface for the user can login and add items are sold themselves.


For now: we are trying on my system with virtual env and can installed cms-django and continuation; download from repository from "mr jrief" django-shop and installed it.  

after: we modify add_intalled and add 'shop'

finallly, I had run
python "manage.py" makemigrations and show this problem:


django.core.exceptions.ImproperlyConfigured: Deferred foreign key 'OrderPayment.order' has not been mapped

"http://django-shop.readthedocs.io/en/latest/reference/deferred-models.html" 

I have reading about astract model and how mapping it, and what about my specify problem, i saw that some people resolve it, but how i can do?, you know i am beginer.  

Whats step can do to having funtionaly django-shp with cms-integrations, and next can development  a module that I told you line before.

**Comment by jrief** [comment]
``OrderPayment`` is added automatically to your project, whenever you register the model admin class inheriting from ``shop.admin.order.BaseOrderAdmin``.

The easiest way to do this is to add to your own ``admin.py``

```
from django.contrib import admin
from shop.admin.defaults.order import OrderAdmin

admin.site.register(Order, OrderAdmin)
```
